ocaml/xapi/tar_ext.ml

```diff
@@ -0,0 +1,118 @@
+(*
+ * Copyright (C) 2006-2009 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Helpers
+
+module D = Debug.Make (struct let name = "tar_ext" end)
+
+open D
+
+let file_perm = 0o644
+
+let dir_perm = 0o755
+
+let dir_size = 4096L
+
+let ( + ) = Int64.add
+
+exception Illegal_file_path of string
```

Comment:
I feel there are a bit of inconsistencies in how we name exception constructors. Considering they are constructors, we should either use `UpperCase` for `Upper_Case`. But I can see there are various existing code that do not adhere to this. @lindig do you have any opinion on this?

Reply:
I have no strong opinion; traditionally OCaml uses underscores but we (and other code) use both conventions.